py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,615 @@
+        kmh_speed = msec_speed * self.kmh_from_msec
+        fsec_speed = msec_speed * self.ft_from_meter
+        mph_speed = fsec_speed * self.mph_from_fsec
+
+        # Gusts
+        msec_gust = wind['gust'] if ('gust' in wind) else 0
+        kmh_gust = msec_gust * self.kmh_from_msec
+        fsec_gust = msec_gust * self.ft_from_meter
+        mph_gust = fsec_gust * self.mph_from_fsec
+
+        # Format the wind speed
+        return self.py3.safe_format(self.format_wind, {
+            'icon': self.icons['wind'],
+            'deg': wind['deg'] if ('deg' in wind) else 0,
+
+            'msec_speed': round(msec_speed),
+            'kmh_speed': round(kmh_speed),
+            'fsec_speed': round(fsec_speed),
+            'mph_speed': round(mph_speed),
+
+            'msec_gust': round(msec_gust),
+            'kmh_gust': round(kmh_gust),
+            'fsec_gust': round(fsec_gust),
+            'mph_gust': round(mph_gust),
+        })
+
+    def _format_humidity(self, wthr):
+        # Format the humidity
+        return self.py3.safe_format(self.format_humidity, {
+            'icon': self.icons['humidity'],
+            'humid': wthr.get_humidity(),
+        })
+
+    def _format_pressure(self, wthr):
+        # Get data and add the icon
+        pressure = wthr.get_pressure()
+        pressure['icon'] = self.icons['pressure']
+
+        # Format the barometric pressure
+        return self.py3.safe_format(self.format_humidity, pressure)
+
+    def _format_temp(self, wthr):
+        # Get data
+        c_data = wthr.get_temperature(unit='celsius')
+        f_data = wthr.get_temperature(unit='fahrenheit')
+        k_data = wthr.get_temperature()
+
+        # Fix forecasts
+        for group in (c_data, f_data, k_data):
+            if ('temp' not in group):
+                group['temp'] = group['day']
+
+            if ('temp_min' not in group):
+                group['temp_min'] = group['min']
+
+            if ('temp_max' not in group):
+                group['temp_max'] = group['max']
+
+        # Format the temperature
+        return self.py3.safe_format(self.format_temp, {
+            'icon': self.icons['temp'],
+
+            'c': round(c_data['temp']),
+            'c_min': round(c_data['temp_min']),
+            'c_max': round(c_data['temp_max']),
+
+            'f': round(f_data['temp']),
+            'f_min': round(f_data['temp_min']),
+            'f_max': round(f_data['temp_max']),
+
+            'k': round(k_data['temp']),
+            'k_min': round(k_data['temp_min']),
+            'k_max': round(k_data['temp_max']),
+        })
+
+    def _format_sunrise(self, wthr):
+        # Get the ISO 8601 time
+        raw_time = wthr.get_sunrise_time('iso')
+        dt = datetime.datetime.strptime(raw_time, "%Y-%m-%d %H:%M:%S+00")
+
+        # Format the sunrise
+        return self.py3.safe_format(self.format_sunrise, {
+            'icon': self.icons['sunrise'],
+            'strftime': dt.strftime(self.format_sunrise_time),
```

Comment:
See if we can avoid `format_sunrise_time` here. Read `%` in `format_sunrise instead`.
See if we can avoid `format_sunset_time` here. Read `%` in `format_sunset instead`.